src/main/haskell/kore/src/Kore/ASTUtils/AlphaCompare.hs

```diff
@@ -0,0 +1,101 @@
+{-|
+Module      : Kore.ASTUtils.AlphaCompare
+Description : Compare
+Copyright   : (c) Runtime Verification, 2018
+License     : NCSA
+Maintainer  : phillip.harris@runtimeverification.com
+Stability   : experimental
+Portability : portable
+-}
+
+module Kore.ASTUtils.AlphaCompare
+    ( alphaCompare
+    ) where
+
+import Data.List
+
+import Kore.AST.Common
+import Kore.ASTUtils.SmartPatterns
+
+alphaCompare
+    :: (Eq (var level), Eq (var level))
+    => PureMLPattern level var
+    -> PureMLPattern level var
+    -> Bool
+alphaCompare e1 e2 = go [] [] e1 e2
+
+go
+    :: (Eq (var level), Eq (var level))
+    => [var level]
+    -> [var level]
+    -> PureMLPattern level var
+    -> PureMLPattern level var
+    -> Bool
+go ctx1 ctx2 e1 e2 = case (e1, e2) of
```

Comment:
Please document how this function works. It would particularly help:
1. to know what the arguments are, and
1. not to use single-letter variable names everywhere.